shadowsocks-csharp/Util/SystemProxy/RAS.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Shadowsocks.Util.SystemProxy
+{
+
+    enum RasFieldSizeConst
+    {
+        MaxEntryName = 256,
+        MaxPath = 260,
+    }
+
+    [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Auto)]
+    struct RasEntryName
+    {
+        public int dwSize;
+
+        [MarshalAs(UnmanagedType.ByValTStr, SizeConst = RAS.MaxEntryName + 1)]
+        public string szEntryName;
+
+        public int dwFlags;
+
+        [MarshalAs(UnmanagedType.ByValTStr, SizeConst = RAS.MaxPath + 1)]
+        public string szPhonebookPath;
+    }
+    class RAS
+    {
+        public const int MaxEntryName = 256;
+        public const int MaxPath = 260;
+
+        const int ESuccess = 0;
+        const int RasBase = 600;
+        const int EBufferTooSmall = 603;
+
+        [DllImport("rasapi32.dll", CharSet = CharSet.Auto)]
```

Comment:
```
// rasapi32.dll is the Remote Access API (RAS), used by Windows applications to control modem connections.
```

Reply:
rasapi32.dll exists since Windows 98. It is safe for win7 users

Reply:
As I mentioned in commit message, these code are mostly from shadowsocksrr.

Reply:
so we can expect same experience with them.